Linux permissions

<!--
Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

Before creating an issue, make sure you've checked the following:
- You are running the latest released version of k0s
- Make sure you've searched for existing issues, both open and closed
- Make sure you've searched for PRs too, a fix might've been merged already
- You're looking at docs for the released version, `main` branch docs are usually ahead of released versions.
    - Docs for exact released version can be found at https://github.com/k0sproject/k0s/tree/<version>/docs

-->

**Version**
```
$ k0s version
v0.9.0
```
**Platform**
```
$ lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 10 (buster)
Release:        10
Codename:       buster
```
**What happened?**

Debian is setup with umask 027 in `/etc/profile` and `/etc/login.defs`. I have setup `containerd` has described in get starting.

When following get started, the command `server` failed with following folders because of their permissions : 
 - `/var/lib/k0s/pki`  (need `751`)
 - `/var/lib/k0s/pki/etcd` (need `711`)
 - `/var/lib/k0s/bin`  (need `755`)
 - `/var/lib/k0s/manifests`  (need `755`)
 - `/var/lib/k0s/manifests/calico`  (need `755`)
 - `/var/lib/k0s/manifests/calico_init`  (need `755`)
 - `/var/lib/k0s/manifests/helm`  (need `755`)
 - `/var/lib/k0s/manifests/konnectivity`  (need `755`)
 - `/var/lib/k0s/run`  (need `755`)
 - `/var/lib/k0s/run/konnectivity-server` (need `755`)

When following get started, the command `worker` failed with following folders because of their permissions : 
 - `/var/lib/k0s/kubelet`  (need `755`)

**How To Reproduce**
- Set umask 027 in `/etc/profile` and `/etc/login.defs` ;
- Run get started from documentation.

**Expected behavior**

When folder are created by k0s, i expect correct permission needed by k0s.

**Screenshots & Logs**
```
root@master-01:~# k0s server --config /etc/k0s/config.yaml                                                                                                                                                                               [0/0]
Error: directory "/var/lib/k0s/pki" exist, but the permission is 0750. The expected permission is 751                                                                                                                                         
Usage:
  k0s server [join-token] [flags]

Examples:
        Command to associate master nodes:
        CLI argument:
        $ k0s server [join-token]

        or CLI flag:
        $ k0s server --token-file [path_to_file]
        Note: Token can be passed either as a CLI argument or as a flag

Flags:
      --enable-worker       enable worker (default false)
  -h, --help                help for server
      --profile string      worker profile to use on the node (default "default")
      --token-file string   Path to the file containing join-token.

Global Flags:
  -c, --config string            config file (default: ./k0s.yaml)
      --data-dir string          Data Directory for k0s (default: /var/lib/k0s). DO NOT CHANGE for an existing setup, things will break!
  -d, --debug                    Debug logging (default: false)
  -l, --logging stringToString   Logging Levels for the different components (default [kubelet=1,kube-proxy=1,etcd=info,containerd=info,konnectivity-server=1,kube-apiserver=1,kube-controller-manager=1,kube-scheduler=1])

directory "/var/lib/k0s/pki" exist, but the permission is 0750. The expected permission is 751
2020-12-28 17:10:41.580840 I | directory "/var/lib/k0s/pki" exist, but the permission is 0750. The expected permission is 751

root@master-01:~# chmod 0751 /var/lib/k0s/pki

root@master-01:~# k0s -d server --config /etc/k0s/config.yaml
time="2020-12-28 17:19:41" level=debug msg="found local addresses: [100.90.1.50]"
time="2020-12-28 17:19:41" level=info msg="using public address: 100.90.1.50"
time="2020-12-28 17:19:41" level=info msg="using sans: [master.k0s.lan]"
time="2020-12-28 17:19:41" level=info msg="DNS address: 10.96.0.10"
time="2020-12-28 17:19:41" level=info msg="Using storage backend etcd"
time="2020-12-28 17:19:41" level=info msg="initializing Certificates\n"
2020-12-28 17:19:41.023227 I | [INFO] generate received request
2020-12-28 17:19:41.023233 I | [INFO] received CSR
2020-12-28 17:19:41.023237 I | [INFO] generating key: rsa-2048
2020-12-28 17:19:41.089686 I | [INFO] encoded CSR
2020-12-28 17:19:41.091630 I | [INFO] signed certificate with serial number 588695822744600165600702614290013336836317871608
time="2020-12-28 17:19:41" level=debug msg="CA key and cert exists, loading"
time="2020-12-28 17:19:41" level=debug msg="creating certificates"
2020-12-28 17:19:41.091980 I | [INFO] generate received request
2020-12-28 17:19:41.091984 I | [INFO] received CSR
2020-12-28 17:19:41.091987 I | [INFO] generating key: rsa-2048
time="2020-12-28 17:19:41" level=debug msg="found local addresses: [100.90.1.50]"
time="2020-12-28 17:19:41" level=debug msg="creating certificates"
2020-12-28 17:19:41.096198 I | [INFO] generate received request
2020-12-28 17:19:41.096202 I | [INFO] received CSR
2020-12-28 17:19:41.096205 I | [INFO] generating key: rsa-2048
time="2020-12-28 17:19:41" level=debug msg="creating certificates"
2020-12-28 17:19:41.098418 I | [INFO] generate received request
2020-12-28 17:19:41.098422 I | [INFO] received CSR
2020-12-28 17:19:41.098425 I | [INFO] generating key: rsa-2048
time="2020-12-28 17:19:41" level=debug msg="creating certificates"
2020-12-28 17:19:41.105019 I | [INFO] generate received request
2020-12-28 17:19:41.105029 I | [INFO] received CSR
2020-12-28 17:19:41.105033 I | [INFO] generating key: rsa-2048
time="2020-12-28 17:19:41" level=debug msg="creating certificates"
2020-12-28 17:19:41.109893 I | [INFO] generate received request
2020-12-28 17:19:41.109899 I | [INFO] received CSR
2020-12-28 17:19:41.109906 I | [INFO] generating key: rsa-2048
time="2020-12-28 17:19:41" level=debug msg="creating certificates"
2020-12-28 17:19:41.112302 I | [INFO] generate received request
2020-12-28 17:19:41.112306 I | [INFO] received CSR
2020-12-28 17:19:41.112309 I | [INFO] generating key: rsa-2048
time="2020-12-28 17:19:41" level=debug msg="creating certificates"
2020-12-28 17:19:41.128647 I | [INFO] generate received request
2020-12-28 17:19:41.128652 I | [INFO] received CSR
2020-12-28 17:19:41.128655 I | [INFO] generating key: rsa-2048
2020-12-28 17:19:41.148720 I | [INFO] generate received request
2020-12-28 17:19:41.148734 I | [INFO] received CSR
2020-12-28 17:19:41.148764 I | [INFO] generating key: rsa-2048
2020-12-28 17:19:41.322815 I | [INFO] encoded CSR
2020-12-28 17:19:41.324571 I | [INFO] signed certificate with serial number 432157627285046559079026388084869600278340277163
time="2020-12-28 17:19:41" level=debug msg="creating certificates"
2020-12-28 17:19:41.324928 I | [INFO] generate received request
2020-12-28 17:19:41.324933 I | [INFO] received CSR
2020-12-28 17:19:41.324942 I | [INFO] generating key: rsa-2048
2020-12-28 17:19:41.510048 I | [INFO] encoded CSR
2020-12-28 17:19:41.572365 I | [INFO] signed certificate with serial number 335661002115512352122705860767283418386501885791
2020-12-28 17:19:41.572387 I | [INFO] encoded CSR
2020-12-28 17:19:41.574257 I | [INFO] signed certificate with serial number 655091255884732771678870575072481214181201845378
2020-12-28 17:19:41.642431 I | [INFO] encoded CSR
2020-12-28 17:19:41.644226 I | [INFO] signed certificate with serial number 656852840271084839202467702005103074171508573563
2020-12-28 17:19:41.674432 I | [INFO] encoded CSR
2020-12-28 17:19:41.676541 I | [INFO] signed certificate with serial number 461588194963587361660597482338413587169502621332
2020-12-28 17:19:41.688000 I | [INFO] encoded CSR
2020-12-28 17:19:41.689760 I | [INFO] signed certificate with serial number 205743385703734130979359187786482937064938099332
2020-12-28 17:19:41.736842 I | [INFO] encoded CSR
2020-12-28 17:19:41.738647 I | [INFO] signed certificate with serial number 585327828795014781832814856945292543395454178225
2020-12-28 17:19:41.859656 I | [INFO] encoded CSR
2020-12-28 17:19:41.861827 I | [INFO] signed certificate with serial number 246182421720753333589451868776036895396906967114
2020-12-28 17:19:41.971747 I | [INFO] encoded CSR
2020-12-28 17:19:41.973628 I | [INFO] signed certificate with serial number 203696333567117444165074069395405750953791563198
time="2020-12-28 17:19:41" level=info msg="initializing Etcd\n"
time="2020-12-28 17:19:41" level=info msg="initializing APIServer\n"
time="2020-12-28 17:19:41" level=info msg="initializing Konnectivity\n"
time="2020-12-28 17:19:41" level=info msg="initializing Scheduler\n"
time="2020-12-28 17:19:41" level=info msg="initializing ControllerManager\n"
time="2020-12-28 17:19:41" level=info msg="initializing Manager\n"
time="2020-12-28 17:19:41" level=info msg="initializing K0SControlAPI\n"
time="2020-12-28 17:19:41" level=info msg="initializing Component\n"
time="2020-12-28 17:19:41" level=info msg="initializing leaderElector\n"
time="2020-12-28 17:19:41" level=warning msg="Running konnectivity as root: user: unknown user konnectivity-server"
time="2020-12-28 17:19:41" level=info msg="kube client has been init" component=telemetry
time="2020-12-28 17:19:41" level=info msg="segment client has been init" component=telemetry
time="2020-12-28 17:19:41" level=warning msg="Running etcd as root: user: unknown user etcd"
time="2020-12-28 17:19:41" level=warning msg="Running kube-apiserver as root: user: unknown user kube-apiserver"
time="2020-12-28 17:19:41" level=info msg="Staging /var/lib/k0s/bin/kube-apiserver"
time="2020-12-28 17:19:41" level=warning msg="Running kube-controller-manager as root: user: unknown user kube-apiserver"
time="2020-12-28 17:19:41" level=info msg="Staging /var/lib/k0s/bin/kube-controller-manager"
time="2020-12-28 17:19:41" level=warning msg="Running kube-scheduler as root: user: unknown user kube-scheduler"
time="2020-12-28 17:19:41" level=info msg="Staging /var/lib/k0s/bin/kube-scheduler"
Error: failed to create manifest bundle dir /var/lib/k0s/manifests: directory "/var/lib/k0s/manifests" exist, but the permission is 0750. The expected permission is 755
Usage:
  k0s server [join-token] [flags]

Examples:
	Command to associate master nodes:
	CLI argument:
	$ k0s server [join-token]

	or CLI flag:
	$ k0s server --token-file [path_to_file]
	Note: Token can be passed either as a CLI argument or as a flag

Flags:
      --enable-worker       enable worker (default false)
  -h, --help                help for server
      --profile string      worker profile to use on the node (default "default")
      --token-file string   Path to the file containing join-token.

Global Flags:
  -c, --config string            config file (default: ./k0s.yaml)
      --data-dir string          Data Directory for k0s (default: /var/lib/k0s). DO NOT CHANGE for an existing setup, things will break!
  -d, --debug                    Debug logging (default: false)
  -l, --logging stringToString   Logging Levels for the different components (default [kube-apiserver=1,kube-controller-manager=1,kube-scheduler=1,kubelet=1,kube-proxy=1,etcd=info,containerd=info,konnectivity-server=1])

failed to create manifest bundle dir /var/lib/k0s/manifests: directory "/var/lib/k0s/manifests" exist, but the permission is 0750. The expected permission is 755
2020-12-28 17:19:41.974756 I | failed to create manifest bundle dir /var/lib/k0s/manifests: directory "/var/lib/k0s/manifests" exist, but the permission is 0750. The expected permission is 755

root@master-01:~# chmod 0755 /var/lib/k0s/manifests

root@master-01:~# k0s -d server --config /etc/k0s/config.yaml 
time="2020-12-28 17:20:55" level=debug msg="found local addresses: [100.90.1.50]"
time="2020-12-28 17:20:55" level=info msg="using public address: 100.90.1.50"
time="2020-12-28 17:20:55" level=info msg="using sans: [master.k0s.lan]"
time="2020-12-28 17:20:55" level=info msg="DNS address: 10.96.0.10"
time="2020-12-28 17:20:55" level=info msg="Using storage backend etcd"
time="2020-12-28 17:20:55" level=info msg="initializing Certificates\n"
time="2020-12-28 17:20:55" level=debug msg="CA key and cert exists, loading"
time="2020-12-28 17:20:55" level=debug msg="found local addresses: [100.90.1.50]"
time="2020-12-28 17:20:55" level=info msg="initializing Etcd\n"
time="2020-12-28 17:20:55" level=info msg="initializing APIServer\n"
time="2020-12-28 17:20:55" level=info msg="initializing Konnectivity\n"
time="2020-12-28 17:20:55" level=info msg="initializing Scheduler\n"
time="2020-12-28 17:20:55" level=info msg="initializing ControllerManager\n"
time="2020-12-28 17:20:55" level=info msg="initializing Manager\n"
time="2020-12-28 17:20:55" level=info msg="initializing K0SControlAPI\n"
time="2020-12-28 17:20:55" level=info msg="initializing Component\n"
time="2020-12-28 17:20:55" level=info msg="initializing leaderElector\n"
time="2020-12-28 17:20:55" level=warning msg="Running etcd as root: user: unknown user etcd"
time="2020-12-28 17:20:55" level=warning msg="Running kube-controller-manager as root: user: unknown user kube-apiserver"
time="2020-12-28 17:20:55" level=info msg="Staging /var/lib/k0s/bin/kube-controller-manager"
time="2020-12-28 17:20:55" level=warning msg="Running konnectivity as root: user: unknown user konnectivity-server"
time="2020-12-28 17:20:55" level=info msg="kube client has been init" component=telemetry
time="2020-12-28 17:20:55" level=info msg="segment client has been init" component=telemetry
time="2020-12-28 17:20:55" level=warning msg="Running kube-apiserver as root: user: unknown user kube-apiserver"
time="2020-12-28 17:20:55" level=info msg="Staging /var/lib/k0s/bin/kube-apiserver"
time="2020-12-28 17:20:55" level=warning msg="Running kube-scheduler as root: user: unknown user kube-scheduler"
time="2020-12-28 17:20:55" level=info msg="Staging /var/lib/k0s/bin/kube-scheduler"
Error: failed to create etcd cert dir: directory "/var/lib/k0s/pki/etcd" exist, but the permission is 0710. The expected permission is 711
Usage:
  k0s server [join-token] [flags]

Examples:
	Command to associate master nodes:
	CLI argument:
	$ k0s server [join-token]

	or CLI flag:
	$ k0s server --token-file [path_to_file]
	Note: Token can be passed either as a CLI argument or as a flag

Flags:
      --enable-worker       enable worker (default false)
  -h, --help                help for server
      --profile string      worker profile to use on the node (default "default")
      --token-file string   Path to the file containing join-token.

Global Flags:
  -c, --config string            config file (default: ./k0s.yaml)
      --data-dir string          Data Directory for k0s (default: /var/lib/k0s). DO NOT CHANGE for an existing setup, things will break!
  -d, --debug                    Debug logging (default: false)
  -l, --logging stringToString   Logging Levels for the different components (default [kube-proxy=1,etcd=info,containerd=info,konnectivity-server=1,kube-apiserver=1,kube-controller-manager=1,kube-scheduler=1,kubelet=1])

failed to create etcd cert dir: directory "/var/lib/k0s/pki/etcd" exist, but the permission is 0710. The expected permission is 711
2020-12-28 17:20:55.860661 I | failed to create etcd cert dir: directory "/var/lib/k0s/pki/etcd" exist, but the permission is 0710. The expected permission is 711

root@master-01:~# chmod 0711 /var/lib/k0s/pki/etcd

root@master-01:~# k0s -d server --config /etc/k0s/config.yaml 
time="2020-12-28 17:23:27" level=debug msg="found local addresses: [100.90.1.50]"
time="2020-12-28 17:23:27" level=info msg="using public address: 100.90.1.50"
time="2020-12-28 17:23:27" level=info msg="using sans: [master.k0s.lan]"
time="2020-12-28 17:23:27" level=info msg="DNS address: 10.96.0.10"
time="2020-12-28 17:23:27" level=info msg="Using storage backend etcd"
time="2020-12-28 17:23:27" level=info msg="initializing Certificates\n"
time="2020-12-28 17:23:27" level=debug msg="CA key and cert exists, loading"
time="2020-12-28 17:23:27" level=debug msg="found local addresses: [100.90.1.50]"
time="2020-12-28 17:23:27" level=info msg="initializing Etcd\n"
time="2020-12-28 17:23:27" level=info msg="initializing APIServer\n"
time="2020-12-28 17:23:27" level=info msg="initializing Konnectivity\n"
time="2020-12-28 17:23:27" level=info msg="initializing Scheduler\n"
time="2020-12-28 17:23:27" level=info msg="initializing ControllerManager\n"
time="2020-12-28 17:23:27" level=info msg="initializing Manager\n"
time="2020-12-28 17:23:27" level=info msg="initializing K0SControlAPI\n"
time="2020-12-28 17:23:27" level=info msg="initializing Component\n"
time="2020-12-28 17:23:27" level=info msg="initializing leaderElector\n"
time="2020-12-28 17:23:27" level=warning msg="Running etcd as root: user: unknown user etcd"
time="2020-12-28 17:23:27" level=info msg="Staging /var/lib/k0s/bin/etcd"
time="2020-12-28 17:23:27" level=warning msg="Running kube-apiserver as root: user: unknown user kube-apiserver"
time="2020-12-28 17:23:27" level=info msg="Staging /var/lib/k0s/bin/kube-apiserver"
time="2020-12-28 17:23:27" level=warning msg="Running konnectivity as root: user: unknown user konnectivity-server"
time="2020-12-28 17:23:27" level=warning msg="Running kube-scheduler as root: user: unknown user kube-scheduler"
time="2020-12-28 17:23:27" level=info msg="Staging /var/lib/k0s/bin/kube-scheduler"
time="2020-12-28 17:23:27" level=warning msg="Running kube-controller-manager as root: user: unknown user kube-apiserver"
time="2020-12-28 17:23:27" level=info msg="Staging /var/lib/k0s/bin/kube-controller-manager"
time="2020-12-28 17:23:27" level=info msg="kube client has been init" component=telemetry
time="2020-12-28 17:23:27" level=info msg="segment client has been init" component=telemetry
Error: failed to create dir /var/lib/k0s/bin: directory "/var/lib/k0s/bin" exist, but the permission is 0750. The expected permission is 755
Usage:
  k0s server [join-token] [flags]

Examples:
	Command to associate master nodes:
	CLI argument:
	$ k0s server [join-token]

	or CLI flag:
	$ k0s server --token-file [path_to_file]
	Note: Token can be passed either as a CLI argument or as a flag

Flags:
      --enable-worker       enable worker (default false)
  -h, --help                help for server
      --profile string      worker profile to use on the node (default "default")
      --token-file string   Path to the file containing join-token.

Global Flags:
  -c, --config string            config file (default: ./k0s.yaml)
      --data-dir string          Data Directory for k0s (default: /var/lib/k0s). DO NOT CHANGE for an existing setup, things will break!
  -d, --debug                    Debug logging (default: false)
  -l, --logging stringToString   Logging Levels for the different components (default [etcd=info,containerd=info,konnectivity-server=1,kube-apiserver=1,kube-controller-manager=1,kube-scheduler=1,kubelet=1,kube-proxy=1])

failed to create dir /var/lib/k0s/bin: directory "/var/lib/k0s/bin" exist, but the permission is 0750. The expected permission is 755
2020-12-28 17:23:27.466963 I | failed to create dir /var/lib/k0s/bin: directory "/var/lib/k0s/bin" exist, but the permission is 0750. The expected permission is 755

root@master-01:~# chmod 0755 /var/lib/k0s/bin

root@master-01:~# k0s -d server --config /etc/k0s/config.yaml 
time="2020-12-28 17:24:41" level=debug msg="found local addresses: [100.90.1.50]"
time="2020-12-28 17:24:41" level=info msg="using public address: 100.90.1.50"
time="2020-12-28 17:24:41" level=info msg="using sans: [master.k0s.lan]"
time="2020-12-28 17:24:41" level=info msg="DNS address: 10.96.0.10"
time="2020-12-28 17:24:41" level=info msg="Using storage backend etcd"
time="2020-12-28 17:24:41" level=info msg="initializing Certificates\n"
time="2020-12-28 17:24:41" level=debug msg="CA key and cert exists, loading"
time="2020-12-28 17:24:41" level=debug msg="found local addresses: [100.90.1.50]"
time="2020-12-28 17:24:41" level=info msg="initializing Etcd\n"
time="2020-12-28 17:24:41" level=info msg="initializing APIServer\n"
time="2020-12-28 17:24:41" level=info msg="initializing Konnectivity\n"
time="2020-12-28 17:24:41" level=info msg="initializing Scheduler\n"
time="2020-12-28 17:24:41" level=info msg="initializing ControllerManager\n"
time="2020-12-28 17:24:41" level=info msg="initializing Manager\n"
time="2020-12-28 17:24:41" level=info msg="initializing K0SControlAPI\n"
time="2020-12-28 17:24:41" level=info msg="initializing Component\n"
time="2020-12-28 17:24:41" level=info msg="initializing leaderElector\n"
time="2020-12-28 17:24:41" level=warning msg="Running etcd as root: user: unknown user etcd"
time="2020-12-28 17:24:41" level=info msg="Staging /var/lib/k0s/bin/etcd"
time="2020-12-28 17:24:41" level=debug msg="bin/etcd.gz is at offset 20452707"
time="2020-12-28 17:24:41" level=warning msg="Running kube-apiserver as root: user: unknown user kube-apiserver"
time="2020-12-28 17:24:41" level=info msg="Staging /var/lib/k0s/bin/kube-apiserver"
time="2020-12-28 17:24:41" level=debug msg="bin/kube-apiserver.gz is at offset 50713559"
time="2020-12-28 17:24:41" level=warning msg="Running konnectivity as root: user: unknown user konnectivity-server"
time="2020-12-28 17:24:41" level=debug msg="Writing static file: /var/lib/k0s/bin/kube-apiserver"
time="2020-12-28 17:24:41" level=warning msg="Running kube-scheduler as root: user: unknown user kube-scheduler"
time="2020-12-28 17:24:41" level=info msg="Staging /var/lib/k0s/bin/kube-scheduler"
time="2020-12-28 17:24:41" level=debug msg="bin/kube-scheduler.gz is at offset 86629858"
time="2020-12-28 17:24:41" level=debug msg="Writing static file: /var/lib/k0s/bin/kube-scheduler"
time="2020-12-28 17:24:41" level=info msg="kube client has been init" component=telemetry
time="2020-12-28 17:24:41" level=info msg="segment client has been init" component=telemetry
time="2020-12-28 17:24:41" level=debug msg="Writing static file: /var/lib/k0s/bin/etcd"
time="2020-12-28 17:24:41" level=warning msg="Running kube-controller-manager as root: user: unknown user kube-apiserver"
time="2020-12-28 17:24:41" level=info msg="Staging /var/lib/k0s/bin/kube-controller-manager"
time="2020-12-28 17:24:41" level=debug msg="bin/kube-controller-manager.gz is at offset 69305065"
time="2020-12-28 17:24:41" level=debug msg="Writing static file: /var/lib/k0s/bin/kube-controller-manager"
Error: failed to initialize directory /var/lib/k0s/run/konnectivity-server: directory "/var/lib/k0s/run/konnectivity-server" exist, but the permission is 0750. The expected permission is 755
Usage:
  k0s server [join-token] [flags]

Examples:
	Command to associate master nodes:
	CLI argument:
	$ k0s server [join-token]

	or CLI flag:
	$ k0s server --token-file [path_to_file]
	Note: Token can be passed either as a CLI argument or as a flag

Flags:
      --enable-worker       enable worker (default false)
  -h, --help                help for server
      --profile string      worker profile to use on the node (default "default")
      --token-file string   Path to the file containing join-token.

Global Flags:
  -c, --config string            config file (default: ./k0s.yaml)
      --data-dir string          Data Directory for k0s (default: /var/lib/k0s). DO NOT CHANGE for an existing setup, things will break!
  -d, --debug                    Debug logging (default: false)
  -l, --logging stringToString   Logging Levels for the different components (default [konnectivity-server=1,kube-apiserver=1,kube-controller-manager=1,kube-scheduler=1,kubelet=1,kube-proxy=1,etcd=info,containerd=info])

failed to initialize directory /var/lib/k0s/run/konnectivity-server: directory "/var/lib/k0s/run/konnectivity-server" exist, but the permission is 0750. The expected permission is 755
2020-12-28 17:24:43.124639 I | failed to initialize directory /var/lib/k0s/run/konnectivity-server: directory "/var/lib/k0s/run/konnectivity-server" exist, but the permission is 0750. The expected permission is 755

root@master-01:~# chmod 0755 /var/lib/k0s/run

root@master-01:~# k0s -d server --config /etc/k0s/config.yaml
time="2020-12-28 17:26:58" level=debug msg="found local addresses: [100.90.1.50]"
time="2020-12-28 17:26:58" level=info msg="using public address: 100.90.1.50"
time="2020-12-28 17:26:58" level=info msg="using sans: [master.k0s.lan]"
time="2020-12-28 17:26:58" level=info msg="DNS address: 10.96.0.10"
time="2020-12-28 17:26:58" level=info msg="Using storage backend etcd"
time="2020-12-28 17:26:58" level=info msg="initializing Certificates\n"
time="2020-12-28 17:26:58" level=debug msg="CA key and cert exists, loading"
time="2020-12-28 17:26:58" level=debug msg="found local addresses: [100.90.1.50]"
time="2020-12-28 17:26:58" level=info msg="initializing Etcd\n"
time="2020-12-28 17:26:58" level=info msg="initializing APIServer\n"
time="2020-12-28 17:26:58" level=info msg="initializing Konnectivity\n"
time="2020-12-28 17:26:58" level=info msg="initializing Scheduler\n"
time="2020-12-28 17:26:58" level=info msg="initializing ControllerManager\n"
time="2020-12-28 17:26:58" level=info msg="initializing Manager\n"
time="2020-12-28 17:26:58" level=info msg="initializing K0SControlAPI\n"
time="2020-12-28 17:26:58" level=info msg="initializing Component\n"
time="2020-12-28 17:26:58" level=info msg="initializing leaderElector\n"
time="2020-12-28 17:26:58" level=warning msg="Running konnectivity as root: user: unknown user konnectivity-server"
time="2020-12-28 17:26:58" level=info msg="Staging /var/lib/k0s/bin/konnectivity-server"
time="2020-12-28 17:26:58" level=warning msg="Running etcd as root: user: unknown user etcd"
time="2020-12-28 17:26:58" level=warning msg="Running kube-scheduler as root: user: unknown user kube-scheduler"
time="2020-12-28 17:26:58" level=info msg="Staging /var/lib/k0s/bin/etcd"
time="2020-12-28 17:26:58" level=debug msg="Re-use existing file:/var/lib/k0s/bin/etcd"
time="2020-12-28 17:26:58" level=info msg="Staging /var/lib/k0s/bin/kube-scheduler"
time="2020-12-28 17:26:58" level=debug msg="Re-use existing file:/var/lib/k0s/bin/kube-scheduler"
time="2020-12-28 17:26:58" level=warning msg="Running kube-controller-manager as root: user: unknown user kube-apiserver"
time="2020-12-28 17:26:58" level=debug msg="Re-use existing file:/var/lib/k0s/bin/konnectivity-server"
time="2020-12-28 17:26:58" level=info msg="kube client has been init" component=telemetry
time="2020-12-28 17:26:58" level=info msg="segment client has been init" component=telemetry
time="2020-12-28 17:26:58" level=info msg="Staging /var/lib/k0s/bin/kube-controller-manager"
time="2020-12-28 17:26:58" level=debug msg="Re-use existing file:/var/lib/k0s/bin/kube-controller-manager"
time="2020-12-28 17:26:58" level=warning msg="Running kube-apiserver as root: user: unknown user kube-apiserver"
time="2020-12-28 17:26:58" level=info msg="Staging /var/lib/k0s/bin/kube-apiserver"
time="2020-12-28 17:26:58" level=debug msg="Re-use existing file:/var/lib/k0s/bin/kube-apiserver"
time="2020-12-28 17:26:58" level=info msg="starting Certificates"
time="2020-12-28 17:26:58" level=info msg="starting Etcd"
time="2020-12-28 17:26:58" level=info msg="Starting etcd"
time="2020-12-28 17:26:58" level=warning msg="etcd db file(s) already exist, not gonna run join process"
time="2020-12-28 17:26:58" level=info msg="starting etcd with args: [--data-dir=/var/lib/k0s/etcd --listen-client-urls=https://127.0.0.1:2379 --advertise-client-urls=https://127.0.0.1:2379 --client-cert-auth=true --listen-peer-urls=https://100.90.1.50:2380 --initial-advertise-peer-urls=https://100.90.1.50:2380 --name=master-01 --trusted-ca-file=/var/lib/k0s/pki/etcd/ca.crt --cert-file=/var/lib/k0s/pki/etcd/server.crt --key-file=/var/lib/k0s/pki/etcd/server.key --peer-trusted-ca-file=/var/lib/k0s/pki/etcd/ca.crt --peer-key-file=/var/lib/k0s/pki/etcd/peer.key --peer-cert-file=/var/lib/k0s/pki/etcd/peer.crt --log-level=info --peer-client-cert-auth=true --enable-pprof=false]"
time="2020-12-28 17:26:58" level=info msg="starting APIServer"
time="2020-12-28 17:26:58" level=info msg="Starting to supervise" component=etcd
time="2020-12-28 17:26:58" level=debug msg="Waiting for storage backend to report back healthy"
time="2020-12-28 17:26:58" level=info msg="Started succesfully, go nuts" component=etcd
time="2020-12-28 17:26:58" level=info msg="[WARNING] Deprecated '--logger=capnslog' flag is set; use '--logger=zap' flag instead" component=etcd
time="2020-12-28 17:26:58" level=info msg="2020-12-28 17:26:58.881337 I | etcdmain: etcd Version: 3.4.14" component=etcd
time="2020-12-28 17:26:58" level=info msg="2020-12-28 17:26:58.881432 I | etcdmain: Git SHA: Not provided (use ./build instead of go build)" component=etcd
time="2020-12-28 17:26:58" level=info msg="2020-12-28 17:26:58.881466 I | etcdmain: Go Version: go1.13.15" component=etcd
time="2020-12-28 17:26:58" level=info msg="2020-12-28 17:26:58.881487 I | etcdmain: Go OS/Arch: linux/amd64" component=etcd
time="2020-12-28 17:26:58" level=info msg="2020-12-28 17:26:58.881509 I | etcdmain: setting maximum number of CPUs to 2, total number of available CPUs is 2" component=etcd
time="2020-12-28 17:26:58" level=info msg="2020-12-28 17:26:58.881580 N | etcdmain: the server is already initialized as member before, starting as etcd member..." component=etcd
time="2020-12-28 17:26:58" level=info msg="[WARNING] Deprecated '--logger=capnslog' flag is set; use '--logger=zap' flag instead" component=etcd
time="2020-12-28 17:26:58" level=info msg="2020-12-28 17:26:58.881655 I | embed: peerTLS: cert = /var/lib/k0s/pki/etcd/peer.crt, key = /var/lib/k0s/pki/etcd/peer.key, trusted-ca = /var/lib/k0s/pki/etcd/ca.crt, client-cert-auth = true, crl-file = " component=etcd
time="2020-12-28 17:26:58" level=info msg="2020-12-28 17:26:58.881830 C | etcdmain: listen tcp 100.90.1.50:2380: bind: address already in use" component=etcd
time="2020-12-28 17:26:58" level=warning msg="exit status 1" component=etcd
time="2020-12-28 17:26:58" level=info msg="respawning in 5 secs" component=etcd
time="2020-12-28 17:26:58" level=debug msg="checking etcd endpoint for health" component=etcd
time="2020-12-28 17:26:58" level=debug msg="etcd is healthy. closing check" component=etcd
time="2020-12-28 17:26:58" level=info msg="Starting kube-apiserver"
time="2020-12-28 17:26:58" level=info msg="starting Konnectivity"
time="2020-12-28 17:26:58" level=info msg="Starting konnectivity"
time="2020-12-28 17:26:58" level=info msg="Starting to supervise" component=kube-apiserver
time="2020-12-28 17:26:58" level=error msg="failed to start server components: directory \"/var/lib/k0s/manifests/konnectivity\" exist, but the permission is 0750. The expected permission is 755"
time="2020-12-28 17:26:59" level=warning msg="failed to initialize reconcilers manifests saver: directory \"/var/lib/k0s/manifests/calico\" exist, but the permission is 0750. The expected permission is 755"
time="2020-12-28 17:26:58" level=info msg="Starting to supervise" component=konnectivity
time="2020-12-28 17:26:59" level=warning msg="failed to initialize reconcilers manifests saver: directory \"/var/lib/k0s/manifests/calico_init\" exist, but the permission is 0750. The expected permission is 755"
time="2020-12-28 17:26:59" level=warning msg="failed to initialize reconcilers manifests saver: directory \"/var/lib/k0s/manifests/helm\" exist, but the permission is 0750. The expected permission is 755"
time="2020-12-28 17:26:59" level=debug msg="checkpoint recorded" checkpoint=starting-component-init component=performance-timer duration="663.54µs" target=server-start
time="2020-12-28 17:26:59" level=debug msg="checkpoint recorded" checkpoint=finished-component-init component=performance-timer duration=2.387668ms target=server-start
time="2020-12-28 17:26:59" level=info msg="Started succesfully, go nuts" component=kube-apiserver
time="2020-12-28 17:26:59" level=info msg="Started succesfully, go nuts" component=konnectivity
time="2020-12-28 17:26:59" level=debug msg="checkpoint recorded" checkpoint=starting-components component=performance-timer duration=2.489217ms target=server-start
time="2020-12-28 17:26:59" level=debug msg="checkpoint recorded" checkpoint=finished-starting-components component=performance-timer duration=133.588939ms target=server-start
time="2020-12-28 17:26:59" level=debug msg="checkpoint recorded" checkpoint=started-reconcilers component=performance-timer duration=139.614397ms target=server-start
time="2020-12-28 17:26:59" level=info msg="Shutting down k0s server"
panic: close of nil channel

goroutine 1 [running]:
github.com/k0sproject/k0s/pkg/component/server.(*KubeProxy).Stop(0xc000362000, 0xc0004d7230, 0xc000927a68)
	/home/ubuntu/actions-runner/_work/k0s/k0s/pkg/component/server/kubeproxy.go:106 +0x2e
github.com/k0sproject/k0s/cmd.startServer(0x0, 0x0, 0xc8f100, 0xc000927d38)
	/home/ubuntu/actions-runner/_work/k0s/k0s/cmd/server.go:277 +0x11b8
github.com/k0sproject/k0s/cmd.glob..func14(0x2f5f8a0, 0xc0003d4ea0, 0x0, 0x3, 0x0, 0x0)
	/home/ubuntu/actions-runner/_work/k0s/k0s/cmd/server.go:86 +0x99
github.com/spf13/cobra.(*Command).execute(0x2f5f8a0, 0xc0003d4e70, 0x3, 0x3, 0x2f5f8a0, 0xc0003d4e70)
	/home/ubuntu/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:850 +0x47c
github.com/spf13/cobra.(*Command).ExecuteC(0x2f5f0c0, 0xc000132030, 0x2074408, 0xc000927f48)
	/home/ubuntu/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/home/ubuntu/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895
github.com/k0sproject/k0s/cmd.Execute()
	/home/ubuntu/actions-runner/_work/k0s/k0s/cmd/root.go:168 +0x31
main.main()
	/home/ubuntu/actions-runner/_work/k0s/k0s/main.go:39 +0x25

root@master-01:~# chmod 0755 /var/lib/k0s/manifests/{konnectivity,calico,calico_init,helm}

root@master-01:~# k0s -d server --config /etc/k0s/config.yaml
```

**Additional context**

I only make logs for master but worker's only failed on kubelet folder.

--- a/internal/pkg/dir/dir.go
+++ b/internal/pkg/dir/dir.go
@@ -19,8 +19,6 @@ import (
 	"fmt"
 	"os"
 	"os/exec"
-
-	"github.com/k0sproject/k0s/internal/pkg/file"
 )
 
 // IsDirectory check the given path exists and is a directory
@@ -54,12 +52,7 @@ func Init(path string, perm os.FileMode) error {
 	if err := os.MkdirAll(path, perm); err != nil {
 		return err
 	}
-	// Check permissions in case directory already existed
-	if err := file.CheckPathPermissions(path, perm); err != nil {
-		return err
-	}
-
-	return nil
+	return os.Chmod(path, perm)
 }
 
 // Copy copies the content of a folder

--- a/internal/pkg/dir/dir_test.go
+++ b/internal/pkg/dir/dir_test.go
@@ -1,0 +1,49 @@
+package dir
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"golang.org/x/sys/unix"
+)
+
+// CheckPermissions checks the correct permissions
+func checkPermissions(t *testing.T, path string, want os.FileMode) {
+	info, err := os.Stat(path)
+	if err != nil {
+		t.Errorf("%s: %v", path, err)
+		return
+	}
+	got := info.Mode().Perm()
+	if got != want {
+		t.Errorf("%s has permission %o. Expected is %o", path, got, want)
+	}
+}
+
+func TestInit(t *testing.T) {
+	dir, err := os.MkdirTemp("", "testExist")
+	if err != nil {
+		t.Errorf("failed to create temp dir: %v", err)
+		return
+	}
+	defer os.RemoveAll(dir)
+
+	foo := filepath.Join(dir, "foo")
+	err = Init(foo, 0700)
+	if err != nil {
+		t.Errorf("failed to create temp dir foo: %v", err)
+	}
+	checkPermissions(t, foo, 0700)
+
+	oldUmask := unix.Umask(0027)
+	bar := filepath.Join(dir, "bar")
+	err = Init(bar, 0755)
+	if err != nil {
+		t.Errorf("failed to create temp dir bar: %v", err)
+	}
+	checkPermissions(t, bar, 0755)
+
+	_ = unix.Umask(oldUmask)
+
+}

--- a/internal/pkg/file/file.go
+++ b/internal/pkg/file/file.go
@@ -19,11 +19,8 @@ package file
 import (
 	"fmt"
 	"os"
-	"runtime"
 
 	"github.com/k0sproject/k0s/internal/pkg/users"
-
-	"github.com/sirupsen/logrus"
 )
 
 // Exists checks if a file exists and is not a directory before we
@@ -34,23 +31,6 @@ func Exists(fileName string) bool {
 		return false
 	}
 	return !info.IsDir()
-}
-
-// CheckPathPermissions checks the correct permissions are for a path (file or directory)
-func CheckPathPermissions(path string, perm os.FileMode) error {
-	dirInfo, err := os.Stat(path)
-	if err != nil {
-		return err
-	}
-	dirMode := dirInfo.Mode().Perm()
-	if dirMode != perm {
-		if runtime.GOOS != "windows" {
-			return fmt.Errorf("directory %q exist, but the permission is %#o. The expected permission is %o", path, dirMode, perm)
-		}
-		logrus.Warnf("directory %q exist, but the permission is %#o. The expected permission is %o", path, dirMode, perm)
-		return nil
-	}
-	return nil
 }
 
 // Chown changes file/dir mode

--- a/inttest/common/footloosesuite.go
+++ b/inttest/common/footloosesuite.go
@@ -68,6 +68,7 @@ type FootlooseSuite struct {
 	KubeAPIExternalPort   int
 	WithLB                bool
 	WorkerCount           int
+	ControllerUmask       int
 
 	ExtraVolumes  []config.Volume
 	tearDownTimer *time.Timer
@@ -398,8 +399,12 @@ func (s *FootlooseSuite) InitController(idx int, k0sArgs ...string) error {
 		return err
 	}
 	defer ssh.Disconnect()
+	umaskCmd := ""
+	if s.ControllerUmask != 0 {
+		umaskCmd = fmt.Sprintf("umask %d;", s.ControllerUmask)
+	}
 	// Allow any arch for etcd in smokes
-	startCmd := fmt.Sprintf("ETCD_UNSUPPORTED_ARCH=%s nohup %s controller --debug %s >/tmp/k0s-controller.log 2>&1 &", runtime.GOARCH, s.K0sFullPath, strings.Join(k0sArgs, " "))
+	startCmd := fmt.Sprintf("%s ETCD_UNSUPPORTED_ARCH=%s nohup %s controller --debug %s >/tmp/k0s-controller.log 2>&1 &", umaskCmd, runtime.GOARCH, s.K0sFullPath, strings.Join(k0sArgs, " "))
 	_, err = ssh.ExecWithOutput(startCmd)
 	if err != nil {
 		s.T().Logf("failed to execute '%s' on %s", startCmd, controllerNode)

--- a/inttest/singlenode/singlenode_test.go
+++ b/inttest/singlenode/singlenode_test.go
@@ -65,6 +65,7 @@ func TestSingleNodeSuite(t *testing.T) {
 	s := SingleNodeSuite{
 		common.FootlooseSuite{
 			ControllerCount: 1,
+			ControllerUmask: 027,
 		},
 	}
 	suite.Run(t, &s)

--- a/pkg/applier/stackapplier.go
+++ b/pkg/applier/stackapplier.go
@@ -61,7 +61,7 @@ func NewStackApplier(path string, kubeClientFactory kubernetes.ClientFactoryInte
 
 // Start both the initial apply and also the watch for a single stack
 func (s *StackApplier) Start() error {
-	debouncer := debounce.New(5*time.Second, s.fsWatcher.Events, func(arg fsnotify.Event) {
+	debouncer := debounce.New(1*time.Second, s.fsWatcher.Events, func(arg fsnotify.Event) {
 		s.log.Debug("debouncer triggering, applying...")
 		err := retry.OnError(retry.DefaultRetry, func(err error) bool {
 			return true

--- a/pkg/debounce/debounce.go
+++ b/pkg/debounce/debounce.go
@@ -66,8 +66,10 @@ func (d *debouncer) Start() {
 	for {
 		select {
 		case event := <-d.input:
-			item = &event
-			ticker.Reset(d.interval)
+			if event.Op != fsnotify.Chmod {
+				item = &event
+				ticker.Reset(d.interval)
+			}
 		case <-ticker.C:
 			if item != nil {
 				d.callback(*item)

